### PR TITLE
Make max_exemplars configurable

### LINKED ISSUE #4791: Add support to configure the property max_exemplars
**What is missing?**
Add support to configure the property `max_exemplars`. 
https://prometheus.io/docs/prometheus/latest/configuration/configuration/#exemplars
```
storage:
  exemplars:
    max_exemplars: 10000000
```

**Why do we need it?**
To increase the max allowed exemplars.

### PR DESCRIPTION
**Pull request by faustodavid**
## Description

This creates a new StorageSettings that allows configuring the max_exemplars

Fixes #4791

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry
Creates a new StorageSettings that allows configuring the max_exemplars

```release-note
Creates a new StorageSettings that allows configuring the max_exemplars
https://prometheus.io/docs/prometheus/latest/configuration/configuration/#exemplars
```

### CODE REVIEW

File: pkg/apis/monitoring/v1/types.go [hunk 1]
```diff
@@ -474,6 +474,17 @@ type PrometheusSpec struct {
 	// Note that if TargetLimit is lower, that value will be taken instead,
 	// except if either value is zero, in which case the non-zero value will be
 	// used.  If both values are zero, no limit is enforced.
+
+	// Storage related settings that are runtime reloadable.
+	StorageSettings *StorageSettingsSpec `json:"storageSettingsSpec,omitempty"`
```
Comment (simonpasquier) [inline pkg/apis/monitoring/v1/types.go:479]: I'm not sure that we should replicate the Prometheus configuration format. We can probably avoid the `storageSettingsSpec` key.

```
spec:
  examplars:
    maxSize: 10000
```

**Review by simonpasquier — COMMENTED** [review]
The operator needs to check that the version of Prometheus supports the examplars configuration. Also what happens if you try to configure examplars storage but forget to pass the exemplar-storage feature flag?

File: pkg/apis/monitoring/v1/types.go (hunk 1 above)
Comment (faustodavid) [reply pkg/apis/monitoring/v1/types.go:479]: I will rename it :)

**Comment by faustodavid** [comment]
> The operator needs to check that the version of Prometheus supports the examplars configuration. Also what happens if you try to configure examplars storage but forget to pass the exemplar-storage feature flag?

If we don't set the  `exemplar-storage` feature flag then Prometheus ignores this setting. If we check the metric `prometheus_tsdb_exemplar_max_exemplars` it will be 0.

**Comment by faustodavid** [comment]
should I add the check for the flag @simonpasquier ?

File: pkg/apis/monitoring/v1/types.go (hunk 1 above)
Comment (faustodavid) [reply pkg/apis/monitoring/v1/types.go]: I miss read your last comment @simonpasquier , so you want it to be:

```yaml
PrometheusSpec:
    exemplars:
        max_exemplars: 10000
```

What if they add new settings in the storage? 

If this is a blocker I will update the PR right now :)

**Pull request closed without merging**